src/Task/SymfonyConsole.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+declare(strict_types=1);
+
+namespace GrumPHP\Task;
+
+use GrumPHP\Formatter\ProcessFormatterInterface;
+use GrumPHP\Runner\TaskResult;
+use GrumPHP\Runner\TaskResultInterface;
+use GrumPHP\Task\Config\ConfigOptionsResolver;
+use GrumPHP\Task\Context\ContextInterface;
+use GrumPHP\Task\Context\GitPreCommitContext;
+use GrumPHP\Task\Context\RunContext;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+/** @extends AbstractExternalTask<ProcessFormatterInterface> */
+class SymfonyConsole extends AbstractExternalTask
+{
+    public static function getConfigurableOptions(): ConfigOptionsResolver
+    {
+        return ConfigOptionsResolver::fromOptionsResolver(
+            (new OptionsResolver())
+                ->setDefaults([
+                    'bin' => './bin/console',
+                    'command' => [],
+                ])
+                ->addAllowedTypes('command', ['string[]'])
+                ->setRequired('command')
+        );
+    }
+
+    public function canRunInContext(ContextInterface $context): bool
+    {
+        return ($context instanceof GitPreCommitContext || $context instanceof RunContext);
+    }
+
+    public function run(ContextInterface $context): TaskResultInterface
+    {
+        $config = $this->getConfig()->getOptions();
```

Comment:
There is currently no way to tell when a task should be ran.
Maybe it's a good idea to include something like:

```php
        $files = $context->getFiles()
            ->extensions($config['triggered_by'])
            ->paths($config['whitelist_patterns'] ?? [])
            ->notPaths($config['ignore_patterns'] ?? []);

        if (0 === \count($files)) {
            return TaskResult::createSkipped($this, $context);
        }
```

It might be handy though, to have an `run_always` option for commands where if it makes no sense to filter the files?

Reply:
True, I had in mind to use the metadata  `enabled`,
but `white/ignore patterns` and `run_always` is more flexible